### PR TITLE
Select Team list container should scale in width based on browser window width #14197

### PR DESCRIPTION
**Pull request by sridhar02**
#### Summary
changes done for the issues mentioned in the ticket

#### Ticket Link

  Fixes https://github.com/mattermost/mattermost-server/issues/14197
  JIRA https://mattermost.atlassian.net/browse/MM-23749

#### Related Pull Requests

- Has server changes (please link here)
- Has redux changes (please link here)
- Has mobile changes (please link here)

#### Screenshots

* overflow-y addition on the mobile view 
image

* Before the change when the max-width was 400px 

image

* after the max-width is set to 800px

image

### CODE REVIEW

**Comment by esethna** [comment]
Thanks @sridhar02, looks good from my end! Adding @andrewbrown00 for a UX review

**Comment by jgilliam17** [comment]
I noticed login screen email/password fields width has changed on the test server.
Is this intentional? 
Screen Shot 2020-07-10 at 12 16 30 PM

**Comment by sridhar02** [comment]
@jgilliam17,actullay it was mentioned in the task to change the width of the teams to 800px, I have changed it and it is effecting the sign-in also as they are using the same styles for both the teams and the sign-in pages.
image

**Comment by andrewbrown00** [comment]
@sridhar02 thanks for the contribution here 👍  

Changes to the `Teams` page looks good ✅ 

image

**Item to resolve**
- [ ] Revert the `Login` page back the previous width, it's too wide

Current | Expected (previous)
----- | -----
image | image

**Comment by jgilliam17** [comment]
Thank you @sridhar02 
Team list page looks good.

@andrewbrown00 found two more instance where field width was increased. Should this also be reverted? Team URL screen actually looks better with the wider field, where most of the url is visible. 

Create a Team 

Screen Shot 2020-07-14 at 10 31 29 AM

Team URL screen 

Screen Shot 2020-07-14 at 10 32 12 AM

**Comment by andrewbrown00** [comment]
> @andrewbrown00 found two more instance where field width was increased. Should this also be reverted? Team URL screen actually looks better with the wider field, where most of the url is visible.

Thanks @jgilliam17, yes, all non-team page instances should be reverted 🙂

**Comment by sridhar02** [comment]
@jgilliam17 @andrewbrown00 I have reverted the changes and made changes that will be applicable to the team container u can check it.

**Comment by andrewbrown00** [comment]
> @jgilliam17 @andrewbrown00 I have reverted the changes and made changes that will be applicable to the team container u can check it.

Looks good from my end. Thanks for quick updates 🙂 , I'll let @jgilliam17 do her review too.

File: components/select_team/select_team.jsx [hunk 1]
```diff
@@ -336,7 +336,7 @@ export default class SelectTeam extends React.PureComponent {
                 {headerButton}
                 <div className='col-sm-12'>
                     <div
-                        className={'signup-team__container'}
+                        className={'select-team__container signup-team__container'}
```
Comment (calebroseland) [inline components/select_team/select_team.jsx:339]: Thanks for isolating, I've been thinking we need a unique class to distinguish some of these screens.

File: sass/responsive/_mobile.scss [hunk 2]
```diff
@@ -1851,6 +1851,11 @@
     .signup-team-all {
         max-height: calc(75vh - (118px + 187px));
     }
+    
+    .signup-team-all .infinite-scroll {
+        overflow-y: auto;
+    }
```
Comment (calebroseland) [inline sass/responsive/_mobile.scss:1857]: This selector doesn't seem to hit the element I presume that it targets.

```html
<div class="infinite-scroll signup-team-all">
```
?
```css
.signup-team-all.infinite-scroll {
    overflow-y: auto;
}
```

digging into this further, though.. I don't know that this is even needed, as the following already kicks in on hover inside (e.g. you start to drag on mobile)
```css
.infinite-scroll:hover {
    overflow-y: scroll;
    overflow-y: overlay;
}
```

Comment (sridhar02) [reply sass/responsive/_mobile.scss]: yes, I have not checked in that way, I think it working on the hover, so I deleted this class now 
```js
.signup-team-all.infinite-scroll {
    overflow-y: auto;
}
```

**Comment by sudheerDev** [comment]
Thank you @sridhar02

**Pull request merged**